manifests/init.pp

```diff
@@ -71,48 +71,57 @@
   Boolean                    $named_autoconf            = true,
   Boolean                    $caching                   = true,
   Boolean                    $use_nmcli                 = false,
-  Optional[String]           $nmcli_device_name         = undef,
+  Optional[String]           $nmcli_connection_name     = undef,
```

Comment:
This parameter name change requires a version bump.

It doesn't need to be a major bump―the resolve module is major version 0 (which SemVer considers it [initial development](https://semver.org/#spec-item-4).  However, it's an API-breaking change to a published module (0.4.0) and introduces new functionality, so I'd suggest 0.5.0.